content/blog/2022/2022-02-14-spring-shell.md

```diff
@@ -0,0 +1,295 @@
+## Spring Shell
+[Spring Shell](https://spring.io/projects/spring-shell) allows us to build a command line application (Shell) using Spring's framework and all the advantages it provides.
+
+## What Is a Shell Anyways?
+A shell provides us with an interface to a system (usually an OS) to which we give commands and parameters. The shell in turn does some useful tasks for us and provides an output.
+
+## Let's Create a Basic Shell
+Since it's a Spring application, our main method has to be annotated with `@SpringBootApplication`
+```java
+@SpringBootApplication  
+public class SpringShellApplication {  
+   public static void main(String[] args) {  
+      SpringApplication.run(SpringShellApplication.class, args);  
+   }  
+}
+```
+
+Now, let's create our first shell command which simulates an SSH:
+```java
+@ShellComponent  
+public class SSHCommand {  
+    Logger log = Logger.getLogger(SSHCommand.class.getName());  
+  
+    @ShellMethod(value = "connect to remote server")  
+    public void ssh(@ShellOption(value = "-s") String remoteServer)  
+    {  
+        log.info(format("Logged to machine '%s'", remoteServer));  
+    }  
+}
+```
+
+The annotation `@ShellComponent` as defined by the Spring team:
+
+> Indicates that an annotated class may contain shell methods (themselves annotated with ShellMethod) that is, methods that may be invoked reflectively by the shell
+
+As for the `@ShellMethod` annotation:
+
+    Used to mark a method as invokable via Spring Shell.
+We can also see the `value` property which is used to describe the command.
+
+The `@ShellOption` annotation simply states that this command takes a parameter named `-s`.
+
+So as a result when we run the application we get a shell that has a command called `ssh` which takes a parameter `-s` and all it does is logging the passed parameter value to the command line.
+
+```bash
+shell:>ssh -s my-machine
+2022-02-11 15:44:04.065  INFO 5648 --- [           main] j.t.springshell.command.SSHCommand       : Logged to machine 'my-machine'
+shell:>
+```
+
+## Deeper Look Into Spring Shell
+### Command Name
```

Comment:
Try to go over your section headers and make them more "active" (containing a verb). I.e. "Modifying the Command Name" or "Working with Parameters", or "Naming Parameters". That makes the table of contents much more engaging.